Rematch/UI.lua

```diff
@@ -186,74 +155,93 @@ function Addon:OnEnable()
+                fx = fx + move(footnotes.Close, 21)
+
+                local footnoteWidth = fx + 4
+                local panelWidth = panel.maxWidth or 280
+
+                footnotes:SetWidth(footnoteWidth)
+                footnotes:Show()
+                panel:SetWidth(panelWidth-footnoteWidth-3)
+            else
+                script:Hide()
+            end
+        end)
+    end
+
+    if rematchVersion < ns.Version:New(5, 0, 0, 0) then
+        self:HookScript(RematchJournal, 'OnShow', function(self)
+            CollectionsJournal:SetAttribute('UIPanelLayout-width', 870)
+            UpdateUIPanelPositions()
+        end)
+        self:HookScript(RematchJournal, 'OnHide', function(self)
+            CollectionsJournal:SetAttribute('UIPanelLayout-width', 710)
+            UpdateUIPanelPositions()
+        end)
     end
 
-    self:SecureHook(RematchLoadedTeamPanel, 'Update', function(panel)
-        local footnotes = panel.Footnotes
-        local script = scriptButtons[footnotes]
-
-        if self:GetScript(RematchSettings.loadedTeam) then
-            script.key = RematchSettings.loadedTeam
-            script:Show()
-            script:ClearAllPoints()
-            script:SetPoint('LEFT', 5, -0.5)
-
-            local fx = 5 + 21
-            fx = fx + move(footnotes.Preferences, 21)
-            fx = fx + move(footnotes.Notes, 21)
-            fx = fx + move(footnotes.WinRecord, footnotes.WinRecord:GetWidth())
-            fx = fx + move(footnotes.Maximize, 21)
-            fx = fx + move(footnotes.Close, 21)
-
-            local footnoteWidth = fx + 4
-            local panelWidth = panel.maxWidth or 280
-
-            footnotes:SetWidth(footnoteWidth)
-            footnotes:Show()
-            panel:SetWidth(panelWidth-footnoteWidth-3)
-        else
-            script:Hide()
-        end
-    end)
-
-    self:HookScript(RematchJournal, 'OnShow', function(self)
-        CollectionsJournal:SetAttribute('UIPanelLayout-width', 870)
-        UpdateUIPanelPositions()
-    end)
-    self:HookScript(RematchJournal, 'OnHide', function(self)
-        CollectionsJournal:SetAttribute('UIPanelLayout-width', 710)
-        UpdateUIPanelPositions()
-    end)
+    self.updateFrames()
 end
 
-function Addon:OnDisable()
-    tDeleteItem(Rematch:GetMenu('TeamMenu'), teamMenu)
-end
+function RematchPlugin:TeardownUI()
+    local rematchVersion = ns.Version:Current('Rematch')
 
-function Addon:FindMenuItem(menu, text)
-    for i, v in ipairs(menu) do
-        if v.text == text then
-            return v
+    self:UnregisterMessage('PET_BATTLE_SCRIPT_SCRIPT_LIST_UPDATE')
+
+    if rematchVersion >= ns.Version:New(5, 0, 0, 0) then
+        tDeleteItem(Rematch.menus:GetDefinition('TeamMenu'), scriptMenuItem)
+        tDeleteItem(Rematch.menus:GetDefinition('LoadedTeamMenu'), scriptMenuItem)
+
+        -- TODO: Currently can't unregister badges
+        -- Rematch.badges:RegisterBadge('teams', 'PetBattleScripts')
```

Comment:
Nice to have [if Gello implements it](https://www.warcraftpets.com/community/forum/viewtopic.php?p=149582#p149582).